### PR TITLE
Bump to .net5 and add audio support based on Merriam Webster

### PR DESCRIPTION
**Pull request by taooceros**

### CODE REVIEW

**Review by harrynull — APPROVED** [review]
Looks awesome! Thank you.

File: src/ECDict.cs [hunk 1]
```diff
@@ -30,7 +30,7 @@ public async Task<Word> QueryAsync(string word, CancellationToken token)
         {
             if (word == "") return null;
 
-            string sql = $"select * from stardict where word = '{word.Replace("'", "''")}'";
+            string sql = $"select * from stardict where word = \"{word.Replace("'", "''")}\"";
```
[the remark this answers is not in the document]
Comment (taooceros) [reply src/ECDict.cs:33]: Ah sorry I pick the wrong commit when I rebase. Let's revert it back.

**Comment by taooceros** [comment]
In addition I change the view model based on Binding, so don't need to manually update value.
Also, save is changed to isavable, which will be called once setting window be closed, or flow being closed. So probably better than save at every single setting change.

**Comment by taooceros** [comment]
Do you think we should fallback to a source audio when the variant audio is not available? Like some adverb may not have separate audio, and the variance of plural may not have separate audio.

**Comment by taooceros** [comment]
https://www.happypre.com/download-mp3-audio-pronunciation-file-of-any-word-in-english/

I find a more comprehensive api from google that looks like cover more words. I will take a try on that later.

**Comment by harrynull** [comment]
> Do you think we should fallback to a source audio when the variant audio is not available? Like some adverb may not have separate audio, and the variance of plural may not have separate audio.

That might be a good idea. The only concern I have is that in some cases it could be misleading/confusing when the pronounciation differs from the actual word.

**Comment by taooceros** [comment]
> That might be a good idea. The only concern I have is that in some cases it could be misleading/confusing when the pronounciation differs from the actual word.

yeah I also have concern about that. The google api may not be available in china, so probably we still need the webster audio as a fallback. I will add an option for that.

**Comment by harrynull** [comment]
> yeah I also have concern about that. The google api may not be available in china, so probably we still need the webster audio as a fallback. I will add an option for that.

I think gstatic is not blocked in China. It resolves to a server in Beijing.

**Comment by taooceros** [comment]
Sorry for the late reply.

> > yeah I also have concern about that. The google api may not be available in china, so probably we still need the webster audio as a fallback. I will add an option for that.
> 
> I think gstatic is not blocked in China. It resolves to a server in Beijing.

That will be nice. Are you in China now? Could you take a try whether it works? We probably can use the webster audio as a backup if gstatic is blocked.

**Comment by harrynull** [comment]
@taooceros I was in China when I replied, and I did test that it worked.

**Comment by harrynull** [comment]
@taooceros I don't think there is a huge need for that, although of course it's always nicer to have a backup.

**Comment by taooceros** [comment]
> @taooceros I don't think there is a huge need for that, although of course it's always nicer to have a backup.

ok hhh, we can leave the code there. If one day it is blocked, we can bring a backup easy. I will delete the ui code later.

**Review by harrynull — APPROVED** [review]
Nice! Thanks

**Comment by taooceros** [comment]
> Nice! Thanks

You are welcome! I really enjoy the plugin.

**Comment by harrynull** [comment]
I invited you as a [collaborator](https://github.com/harrynull/Flow.Launcher.Dictionary/invitations) and feel free to merge the pr whenever.

**Comment by harrynull** [comment]
> > Nice! Thanks
> 
> You are welcome! I really enjoy the plugin.

Glad you like it! It always makes my day when something I made helps others.
And thank you a lot for your contributions to Flow Launcher. I use it daily and it has been great!

**Comment by taooceros** [comment]
I add a small fix to check whether the query is chinese earlier than the length check so single chinese character will be translated. It's a fairly simple change so I think it's ok to add that without review.

**Pull request merged**